Shared/Samples/Download vector tiles to local cache/DownloadVectorTilesToLocalCacheView.swift

```diff
@@ -77,7 +92,7 @@ struct DownloadVectorTilesToLocalCacheView: View {
                             Button("Download Vector Tiles") {
                                 isDownloading = true
                             }
-                            .disabled(model.isDownloadDisabled || isDownloading)
+                            .disabled(model.exportVectorTilesTask == nil || isDownloading)
```

Comment:
This is an edge case but I can reliably crash the app here if I tap the button before the preview tiles have loaded. Is there another condition that we can use to disable this button until the preview has fully loaded?

Reply:
Are you using an iPad simulator to test? I made a small change in af8f130 , let me know if it solves the problem. vector tile download is super slow at my end… 🐌

Reply:
Hmm, not sure. I think I may have been on a simulator. I just tried to reproduce with the latest updates on my physical iPad and crashed it here. 
<img width="740" alt="Screenshot 2023-01-27 at 16 33 13" src="https://user-images.githubusercontent.com/16397058/215230153-2cc97193-1e00-4b40-8ff1-04b212864e79.png">

Reply:
Thank you for reporting this. I made a commit a2b7028 as a quick patch. There were a few things wrong with this sample

- No safeguard for the export vector tiles task load status. Previously the task is directly assigned to the model's instance property before it is loaded, which allows the button to crash the sample
- Instead of using the `model.exportVectorTilesTask == nil` as the condition, a better one could be `model.exportVectorTilesTask.loadStatus == LoadStatus.loaded`. After all it doesn't make sense to allow using tapping the button before the task is loaded. I need to test if the task's load status is also published and can trigger the UI recompute though.
- If `vectorTileSourceInfo.allowsExportingTiles` is false, the download shouldn't be started. I'll break the `Model.downloadVectorTiles(extent:)` method into 2 part: condition check and async throw download. So no UI would be changed if the download task doesn't allow download in the first place.

Since they are not related to the view model change, I'll fix them in a separate PR. Issue logged at #114